### PR TITLE
slideshow: enable first slide transition

### PR DESCRIPTION
**Pull request by codewithvk**
Change-Id: Ib92f50176d30ae46b6ef9a4cfc7a62c736170a23

* Resolves: # 
* Target version: master

### CODE REVIEW

File: browser/src/slideshow/SlideShowPresenter.ts [hunk 1]
```diff
@@ -247,12 +247,34 @@ class SlideShowPresenter {
...
-		const blankTexture = this._slideRenderer.createEmptyTexture(
-			this._presentationInfo.docWidth,
-			this._presentationInfo.docHeight,
-		);
+		const slideInfo = this.getSlideInfo(this._currentSlide);
+
+		// To speed up the process, if we have transition info, then only render
+		// a black empty slide as the first slide. otherwise, directly render the first slide.
+		if (
+			slideInfo?.transitionType != undefined &&
```
Comment (codewithvk) [inline browser/src/slideshow/SlideShowPresenter.ts:255]: @quikee Should we add more checks here for animation/video?

File: browser/src/slideshow/RenderContext.ts [hunk 2]
```diff
@@ -75,6 +82,39 @@ class RenderContextGl extends RenderContext {
...
+		height: number,
+	): WebGLTexture | ImageBitmap {
+		const gl = this.getGl();
+		const texture = gl.createTexture();
+		if (!texture) {
+			throw new Error('Failed to create texture');
+		}
+		gl.bindTexture(gl.TEXTURE_2D, texture);
+
+		const blackPixels = new Uint8Array(width * height * 4).fill(0);
```
Comment (codewithvk) [inline browser/src/slideshow/RenderContext.ts:96]: Note to my self: Try  new Uint8Array([0, 0, 0, 255]); instade of height width ration. Will be faster.

File: browser/src/slideshow/SlideShowPresenter.ts (hunk 1 above)
Comment (quikee) [reply browser/src/slideshow/SlideShowPresenter.ts:255]: I don't see the need..

**Review by quikee — APPROVED** [review]
LOoks good to me

**Pull request merged**